security/auth_providers.rst

```diff
@@ -1,23 +1,88 @@
-.. index::
-   single: Security; Pre authenticated providers
+Built-in Authentication Providers
+=================================
 
-Using pre Authenticated Security Firewalls
-==========================================
+If you need to add authentication to your app, we recommend using
+:doc:`Guard authentication </security/guard_authentication>` because it gives you
+full control over the process.
 
-A lot of authentication modules are already provided by some web servers,
-including Apache. These modules generally set some environment variables
-that can be used to determine which user is accessing your application. Out of the
-box, Symfony supports most authentication mechanisms.
-These requests are called *pre authenticated* requests because the user is already
-authenticated when reaching your application.
+But, Symfony also offers a number of built-in authentication providers: systems
+that are easier to implement, but harder to customize. If your authentication
+use-case matches one of these exactly, they're a great option:
 
-.. caution::
+.. toctree::
+    :hidden:
 
-    :doc:`User impersonation </security/impersonating_user>` is not
-    compatible with pre-authenticated firewalls. The reason is that
-    impersonation requires the authentication state to be maintained server-side,
-    but pre-authenticated information (``SSL_CLIENT_S_DN_Email``, ``REMOTE_USER``
-    or other) is sent in each request.
+    form_login
+    json_login_setup
+
+* :doc:`form_login </security/form_login>`
+* :ref:`http_basic <security-http_basic>`
+* :doc:`LDAP via HTTP Basic or Form Login </security/ldap>`
+* :doc:`json_login </security/json_login_setup>`
+* :ref:`X.509 Client Certificate Authentication (x509) <security-x509>`
+* :ref:`REMOTE_USER Based Authentication (remote_user) <security-remote_user>`
+* ``simple_form``
+* ``simple_pre_auth``
+
+.. _security-http_basic:
+
+HTTP Basic Authentication
+-------------------------
+
+To support HTTP Basic authentication, add the ``http_basic`` key to your firewall:
```

Comment:
Could we please add a super concise description of each authentication? For example, for "HTTP basic". Something like this, but better explained and more concise:

```
`HTTP Basic authentication`_ asks credentials (username and password) using a dialog in the browser. The given credentials are included in the HTTP request without any hashing or encryption, so it's recommended to use it with HTTPS. This authentication is rarely used on production, but it can come in handy when developing the application prototype or during the local development.

.. _`HTTP Basic authentication`: https://en.wikipedia.org/wiki/Basic_access_authentication
````